homeassistant/components/squeezebox/media_player.py

```diff
@@ -507,3 +565,40 @@ async def async_call_method(self, command, parameters=None):
             for parameter in parameters:
                 all_params.append(parameter)
         await self.async_query(*all_params)
+
+    async def async_call_query(self, command, parameters=None):
+        """
+        Call Squeezebox JSON/RPC method where we care about the result.
+
+        Additional parameters are added to the command to form the list of
+        positional parameters (p0, p1...,  pN) passed to JSON/RPC server.
+        """
+        all_params = [command]
+        if parameters:
+            for parameter in parameters:
+                all_params.append(parameter)
+        self._query_result = await self.async_query(*all_params)
```

Comment:
We shouldn't extend this integration until the device specific interface code and data parsing has been moved to a standalone library published on pypi.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices

Reply:
Ok, I can do this. It will also mean changing the existing code in the integration. Should I add that to this pull request or create a new one? Is there an existing maintainer for this code I should talk to before making these changes?

Reply:
Please make a separate PR where the existing code is refactored to use the PyPI library. Then after that is merged we can rebase here and continue.

I don't believe there's a codeowner for this integration.

Reply:
Makes sense. Will do.

Reply:
For reference, here is the pull request to refactor the plugin. https://github.com/home-assistant/core/pull/34731

Reply:
I've updated this pull request with changes following the squeezebox component refactor to use external library pysqueezebox.